pkgs/servers/http/nginx/nix-etag-1.15.4.patch

```diff
@@ -46,19 +45,21 @@ index cb49ef74..7b456993 100644
 +            return NGX_ERROR;
 +        }
 +
-+        *ptr2++ = '"';
 +        *ptr2 = '\0';
 +
-+        etag->value.len = ngx_strlen(ptr1);
-+        etag->value.data = ngx_pnalloc(r->pool, etag->value.len);
++        etag->value.data = ngx_pnalloc(r->pool, NGX_OFF_T_LEN + ngx_strlen(ptr1) + 3);
 +
 +        if (etag->value.data == NULL) {
 +            ngx_free(real);
 +            etag->hash = 0;
 +            return NGX_ERROR;
 +        }
 +
-+        ngx_memcpy(etag->value.data, ptr1, etag->value.len);
++        etag->value.len = ngx_sprintf(etag->value.data, "\"\%s-%xO\"",
```

Comment:
I don't understand this line, this is supposed to compute an integer, right? Why are we doing string manipulations?

I think, in general, it's good if this patch contains excellent comments on what is happening, I do not want to ship NGINX with potentially memory unsafe patches.

Reply:
`value.len` is set to the length of the resulting string, and `value.data` is set to the string itself. We're allocating from the request's memory pool (which gets freed after the request is handled), and writing into that allocated memory. The `content-length` part of the string can vary in length, which is why we don't know `value.len` ahead of time.

This is the same logic that the [upstream does](https://github.com/nginx/nginx/blob/b8fb83b8d2e7ca03d43176b767f1fc657f1c1ee2/src/http/ngx_http_core_module.c#L1695-L1704) for Etags, and what the patched Nginx does for paths outside of Nix store.

Reply:
Makes sense!